src/include/rbd/librbd.hpp

```diff
@@ -273,6 +305,51 @@ class CEPH_RBD_API RBD
     void release();
   };
 
+  // This must be dynamically allocated with new, and
+  // must be released with release().
+  // Do not use delete.
+  struct AioGroupCompletion {
+    typedef enum {
+      AIO_STATE_PENDING = 0,
+      AIO_STATE_COMPLETE,
+    } aio_state_t;
+    std::atomic<aio_state_t> m_state{AIO_STATE_PENDING};
```

Comment:
Implementation details like this shouldn't be exposed in a public header file.  We need to follow the pattern of `RBD::AioCompletion` with an opaque `pc` pointer -- this is the PIMPL idiom.

The include of an internal `src/librbd/AsioEngine.h` header above is a red flag for the same.  Note that it's not even under `src/include` directory.

Reply:
Removed RBD::AioGroupCompletion implementation